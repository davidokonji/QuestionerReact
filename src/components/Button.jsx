import React from 'react';

const Button = (props) => {
  const { text, event, styles } = props;
  return (
    <button type="button" onClick={event} className={styles}>
      {text}
    </button>
  );
};

export default Button;
