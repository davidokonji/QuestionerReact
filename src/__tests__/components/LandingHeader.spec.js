import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from '../../components/Landingheader';
import Button from '../../components/Button';

let wrapper;

describe('Landing Header component', () => {
  beforeAll(() => {
    wrapper = shallow(<Header page='landingpage' navClass='landing' />);
  });
  test('should render header component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
