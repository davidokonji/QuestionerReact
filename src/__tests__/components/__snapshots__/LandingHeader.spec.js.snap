// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Header component should render header component correctly 1`] = `
<div>
  <nav>
    <NavLink
      className="header_link nav-title"
      exact={true}
      to="/"
    >
      Questioner
    </NavLink>
    <NavLink
      to="/auth"
    >
      <Button
        styles="button rn_button sm_button"
        text="ADMIN"
      />
    </NavLink>
  </nav>
</div>
`;

exports[`Landing Header component should render header component with props 1`] = `
<div>
  <nav
    className="landing"
  >
    <NavLink
      className="header_link nav-title"
      exact={true}
      to="/"
    >
      Questioner
    </NavLink>
    <NavLink
      to="/auth"
    >
      <img
        alt="user"
        src={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        style={
          Object {
            "marginRight": "3rem",
            "width": "3rem",
          }
        }
      />
    </NavLink>
  </nav>
</div>
`;
